### ISSUE TITLE
dbm-core : masque addon , skin glow issue

### ISSUE THREAD

**Issue by Wagerssi** [DeadlyBossMods/DeadlyBossMods#1340]
<img width="160" alt="화면 캡처 2024-10-22 082559" src="https://github.com/user-attachments/assets/704cedc6-eb9b-4d59-8625-69ca6b14d8df">
<img width="598" alt="화면 캡처 2024-10-22 082652" src="https://github.com/user-attachments/assets/ede110f0-1ea8-4fc9-b13f-fca5ac891148">

If you activate dbm-core as in the attached picture, the glow effect will be weird for icons created by weakauras.
This is the result of testing with dbm/masque/weakauras 3 addons activated.

I don't know why dbm is affecting the glow effect.
This symptom occurred suddenly recently.

**Comment by MysticalOS** [comment]
Unfortunately this seems to be caused by that addon modifying a public library, so because we embed public version of library. the public library is loading over their modified one.

This is something they need to address. I even talked with maintainer Weak Auras and LibGlow directly and they agree. masque shouldn't be modifying a public library other addons use cause stuff like this happens. If they are gonna modify the lib, they should rename/register it as a diff name at very least, so when other addons use the public library their addon isn't broken (or worse, they start breaking other addons cause it's all subjective to load order. only reason their mod is being broken by DBM and not DBM broken bon their mod, is DBM is loading first so our version of libglow is loading before theirs)

TL/DR, tell them to stop modifying a public library, or at very least fork it under a diff name so it won't be broken by other addons using the glow library

**Comment by StormFX** [comment]
Masque doesn't even reference LibButtonGlow, much less utilize a "modified copy". In fact, LibButtonGlow calls Masque directly.

**Comment by MysticalOS** [comment]
them I'm perplexed where breakage is. cause I don't see how we can break glows by simply including LibCustomGlow. multiple users have reported this breaks when enabling DBM masque and weak auras. breakage does not occur if not all 3 are present.

**Comment by StormFX** [comment]
It _might_ be related to LCG reusing the animations. I'm going to look into it and see if I can track it down. I don't _think_ it's DBM causing it, I just need to find a way to mitigate whatever is causing it, assuming I can find it.

**Comment by MysticalOS** [comment]
Thanks. Yeah i know me and stan both confused.  Problem was introduced by dbm adding lib and just using it's apis per it's docs.

I know plater uses the lib too and I don't think that introduced issue so i still think it's a condition that requires libcustomglow loading BEFORE your code does. which didn't occur prior to DBM. because as I pointed out above Plater loads after you, Weak auras loads after you. So magically the issue was avoided UNTIL DBM loaded libcustomglow before you. so issue coulda been in lib all along.

**Comment by StormFX** [comment]
If it's a load order issue, those with LCG as installed separately wouldn't experience the problem because LCG has Masque listed as an OptDep. So if it's because DBM loads first, I'm not sure how to go about fixing it, other than to add a (redundant) OptDep for Masque to DBM's ToC file(s). Let me do some testing later first, though, to be sure.

**Comment by MysticalOS** [comment]
I mean if it's a bug in lib we just make stan fix it. vs trying to hack the load order.

**Comment by StormFX** [comment]
I wouldn't call it a bug, really, but definitely an issue. That said, I tested it. Turns out it IS the load order. Adding Masque to DBM Core's ToC fixed it. Then, I removed it and changed line 13 of `LibCustomGlow-1.0.lua` from:

```lua
local Masque = LibStub("Masque", true)
```
To:
```lua
local Masque = C_AddOns.LoadAddOn("Masque") and LibStub("Masque", true)
```
And that also worked.

**Comment by MysticalOS** [comment]
ANd that has to be added to DBM? that's not an ideal workaround. i'd rather have @Stanzilla figure out what's up with it

**Comment by StormFX** [comment]
No, it needs to be changed in LibCustomGlow's `.lua` file, or at least LCG needs to handle it.

**Comment by StormFX** [comment]
https://github.com/Stanzilla/LibCustomGlow/issues

**Comment by MysticalOS** [comment]
Is Masque something useful for DBM to support? I guess what it does is you inject ability to reshape icons? this include namplate icons? I could maybe just support it in our namplate code if I understand more about it

**Comment by StormFX** [comment]
It's up to you and whether your users would want it. Nameplate icons are a bit iffy, since they're constantly updating, but it's certainly possible. Masque is fairly easy to implement, though the docs are a little dated.

**Comment by Wagerssi** [comment]
Any idea when this issue will be resolved?

**Comment by StormFX** [comment]
Not sure. It has to be done through LibCustomGlow. In the interim, you can add Masque to DBM-Core's ToC file for whichever flavor you're playing.

Eg:

Change:
```
## OptionalDependencies: LibStub, CallbackHandler-1.0, LibSharedMedia-3.0, LibChatAnims, LibDBIcon-1.0, LibDeflate, LibSerialize, LibSpecialization, CustomNames
```
To:
```
## OptionalDependencies: LibStub, CallbackHandler-1.0, LibSharedMedia-3.0, LibChatAnims, LibDBIcon-1.0, LibDeflate, LibSerialize, LibSpecialization, CustomNames, Masque
```

**Comment by Stanzilla** [comment]
I think that is actually the correct solution for the whole problem. we don't want to do any LoadAddon dances in the lib and every addon that uses the lib should have it in its OptDeps

**Comment by MysticalOS** [comment]
For clarification is Masque available on cata or vanilla classic?

**Comment by MysticalOS** [comment]
I went ahead and just added the load dep on every flavor for good measure

**Comment by Wagerssi** [comment]
thank you
After modifying the TOC file, we confirmed that Masque's glow effect operates normally. I hope this applies to the next DBM update as well. (WW/Cata/Classic)

**Comment by StormFX** [comment]
> I think that is actually the correct solution for the whole problem. we don't want to do any LoadAddon dances in the lib and every addon that uses the lib should have it in its OptDeps

Can you add this to LCG's docs? :P

> For clarification is Masque available on cata or vanilla classic?

All three flavors. Retail, Classic Era and <Expac> Classic.

Since MysticalOS added it, it should in the next release.

**Comment by Stanzilla** [comment]
> > I think that is actually the correct solution for the whole problem. we don't want to do any LoadAddon dances in the lib and every addon that uses the lib should have it in its OptDeps
> 
> Can you add this to LCG's docs? :P
> 
> > For clarification is Masque available on cata or vanilla classic?
> 
> All three flavors. Retail, Classic Era and Classic.
> 
> > I hope this applies to the next DBM update as well. (WW/Cata/Classic)
> 
> Since MysticalOS added it, it should in the next release.

Done!

**Comment by StormFX** [comment]
> Done!

Masque needs to be in the OptionalDependencies array, not LCG, AFAIK.

**Comment by Stanzilla** [comment]
> > Done!
> 
> Masque needs to be in the OptionalDependencies array, not LCG, AFAIK.
> 
> > If you want to use LCG in your addon, please make sure to add **Masque** to your OptionalDependencies array in your TOC file.

 brain

**Comment by StormFX** [comment]
Dunno about anyone else, but mine's been fired the last few years. :P
